### PR TITLE
STYLE: Minor style and readability changes

### PR DESCRIPTION
**Pull request by hjmjohnson**
Added explicit [[fallthrough]] annotations to make code easier to understand.

Added space in diagnostic messages to make errors easier to read.

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Modules/Core/Common/include/itkImageToImageFilter.hxx [hunk 1]
```diff
@@ -208,7 +208,7 @@ ImageToImageFilter<TInputImage, TOutputImage>::VerifyInputInformation() const
         {
           directionString.setf(std::ios::scientific);
           directionString.precision(7);
-          directionString << "InputImage Direction: " << inputPtr1->GetDirection() << ", InputImage" << it.GetName()
+          directionString << "InputImage Direction: " << inputPtr1->GetDirection() << ", InputImage " << it.GetName()
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkImageToImageFilter.hxx:211]: For the sake of completeness, you could make similar changes to `ImageSink<TInputImage>::VerifyInputInformation()`, if you want to 🤷

File: Modules/IO/GDCM/src/itkGDCMImageIO.cxx [hunk 2]
```diff
@@ -635,12 +635,19 @@ GDCMImageIO::InternalReadImageInformation()
   switch (ms)
   {
     case gdcm::MediaStorage::HardcopyGrayscaleImageStorage:
+      [[fallthrough]];
     case gdcm::MediaStorage::GEPrivate3DModelStorage:
+      [[fallthrough]];
     case gdcm::MediaStorage::Philips3D:
+      [[fallthrough]];
```
Comment (N-Dekker) [inline Modules/IO/GDCM/src/itkGDCMImageIO.cxx:642]: @hjmjohnson Just wondering, did the original code cause any warning?

**Review by dzenanz — APPROVED** [review]
Good even as-is.

File: Modules/IO/GDCM/src/itkGDCMImageIO.cxx (hunk 2 above)
Comment (N-Dekker) [reply Modules/IO/GDCM/src/itkGDCMImageIO.cxx:642]: The C++ core guideline is to use `[[fallthrough]]` only for non-empty cases: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es78-dont-rely-on-implicit-fallthrough-in-switch-statements

It says:

> Multiple case labels of a single statement is OK:

```cpp
switch (x) {
case 'a':
case 'b':
case 'f':
    do_something(x);
    break;
}
```

**Review by N-Dekker — APPROVED** [review]
Thank you very much for addressing my comments, Hans! 👍 👍

**Review by dzenanz — APPROVED** [review]

File: Modules/IO/GDCM/src/itkGDCMImageIO.cxx (hunk 2 above)
Comment (seanm) [reply Modules/IO/GDCM/src/itkGDCMImageIO.cxx]: Yeah, I wouldn't put any `[[fallthrough]]` on empty cases either.

I don't think the old way would have had any warnings, at least not from clang's `-Wimplicit-fallthrough`.

I just noticed I don't have that flag enabled on my bots... I will check if there are none and enable it...

**Pull request merged**